v2/blockstore/readonly.go

```diff
@@ -22,28 +22,38 @@ import (
 	internalio "github.com/ipld/go-car/v2/internal/io"
 )
 
-var _ blockstore.Blockstore = (*ReadOnly)(nil)
-
-// ReadOnly provides a read-only CAR Block Store.
-type ReadOnly struct {
-	// mu allows ReadWrite to be safe for concurrent use.
-	// It's in ReadOnly so that read operations also grab read locks,
-	// given that ReadWrite embeds ReadOnly for methods like Get and Has.
-	//
-	// The main fields guarded by the mutex are the index and the underlying writers.
-	// For simplicity, the entirety of the blockstore methods grab the mutex.
-	mu sync.RWMutex
-
-	// The backing containing the data payload in CARv1 format.
-	backing io.ReaderAt
-	// The CARv1 content index.
-	idx index.Index
-
-	// If we called carv2.NewReaderMmap, remember to close it too.
-	carv2Closer io.Closer
-
-	ropts carv2.ReadOptions
-}
+var (
+	_                    blockstore.Blockstore = (*ReadOnly)(nil)
+	noopErrHandler                             = func(error) {}
```

Comment:
this should be a proper func declaration, not a func variable. func variables should only be used when the function can be nil or its value isn't static, but neither applies here.